src/client/connect/http.rs

```diff
@@ -612,6 +633,14 @@ fn connect(
     )
     .map_err(ConnectError::m("tcp bind local error"))?;
 
+    #[cfg(any(target_os = "android", target_os = "fuchsia", target_os = "linux"))]
+    // That this only works for some socket types, particularly AF_INET sockets.
+    if config.local_address_ipv6.is_none() && config.interface.is_some() {
```

Comment:
Why only if the ipv6 address is none? What if the ipv4 address is some? What if ipv6 is some?

Reply:
Sorry, I mistakenly thought that IPv6 does not support `SO_BINDTODEVICE` operation. Later,  I tried, is feasible.